dds/DCPS/Service_Participant.cpp

```diff
@@ -435,6 +442,13 @@ Service_Participant::get_domain_participant_factory(int &argc,
       }
 
       this->monitor_.reset(this->monitor_factory_->create_sp_monitor(this));
+
+#ifdef ACE_LINUX
+      network_config_publisher_ = make_rch<LinuxNetworkConfigPublisher>(reactor_task_.interceptor());
```

Comment:
I'd rather see this tied to the choice of Discovery / Transport. Shmem applications aren't going to need to monitor network changes.

Reply:
The scenario I'm targeting is RTPS Discovery and RTPS Transport.  After we add ICE, that means we would need to run three copies of the same service to detect network changes.  Transports and discoveries that don't use it won't.  I can see an argument for lazily creating it.